iceoryx_utils/include/iceoryx_utils/internal/cxx/list.inl

```diff
@@ -323,31 +323,31 @@ template <typename T, uint64_t Capacity>
 inline T& list<T, Capacity>::front() noexcept
 {
     auto iter = begin();
-    handleInvalidElement(iter.m_iterListNodeIdx);
+    IOX_DISCARD_RESULT(handleInvalidElement(iter.m_iterListNodeIdx));
```

Comment:
Could you please remove the `handleInvalidElement` and all the calls and replace it with
```cpp
cxx::Expects(isValidElementIdx(idx));
```
This is what the method does, we do not need to overcomplicate things here...